src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1242,19 +1246,191 @@
+                        ;; Letting the key carry all of the metadata, this makes the batch evaluation of the expressions
+                        ;; easier because the metadata contains unevalable forms.
+                        (with-meta [id expr-key] (assoc metadata expr-key s-expr))
+                        s-expr))
+        is-root? #(= #{0} (get backward-edges %))
+
+        ;; If extract-exprs ever became a hot spot, this could be changed out to use more java interop.
+        alpha-exprs (reduce (fn [prev alpha-node]
+                              (let [{:keys [id condition env]} alpha-node
+                                    {:keys [type constraints fact-binding args]} condition
+                                    cmeta (meta condition)]
+                                (handle-expr prev
+                                             (with-meta (compile-condition
+                                                          type (first args) constraints
+                                                          fact-binding env)
+                                                        ;; Remove all metadata but file and line number
+                                                        ;; to protect from evaluating usafe metadata
+                                                        ;; See PR 243 for more detailed discussion
+                                                        (select-keys cmeta [:line :file]))
+                                             id
+                                             :alpha-expr
+                                             {:file (or (:file cmeta) *file*)
+                                              :compile-ctx {:condition condition
+                                                            :env env
+                                                            :msg "compiling alpha node"}})))
+                            {}
+                            alpha-graph)
+        exprs (reduce-kv (fn [prev id production-node]
+                             (let [production (-> production-node :production)]
+                               (handle-expr prev
+                                            (with-meta (compile-action (:bindings production-node)
+                                                                       (:rhs production)
+                                                                       (:env production))
+                                                       (meta (:rhs production)))
+                                            id
+                                            :action-expr
+                                            ;; ProductionNode expressions can be arbitrary code, therefore we need the
+                                            ;; ns where the production was define so that we can compile the expression
+                                            ;; later.
+                                            {:ns (some-> (:ns-name production) str)
```

Comment:
Why is the `:ns` stored as a string here? Also, I think it more typically to do 
`(some-> (:ns-name production) name)` if a string is desired from a ns sym.

Reply:
Hold over from a previous pass, where this metadata was on the fn that would have been eval'd. Also a reason that 1250 was chosen in the other locations.

The addition of the metadata push the limit well below what we discussed on this comment 
https://github.com/cerner/clara-rules/pull/396#discussion_r197290343